### PR TITLE
Cap 364 fix temporal extent

### PR DESCRIPTION
**Pull request by tomw66**
Covering [this ticket](https://metoffice.atlassian.net/browse/CAP-364) - with some additional changes to the arguments passed to `Parameter` as this was also causing issues when attempting to upload aircraft data.

Will require [this branch](https://github.com/MetOffice/edr_server/pull/35#issue-1422385271) in `edr_server` to also be merged in order to be able to upload aircraft data.

### CODE REVIEW

File: src/clean_air/data/extract_metadata.py [hunk 1]
```diff
@@ -91,24 +95,32 @@ def extract_metadata(
                 spatial_extent = SpatialExtent(bounding_polygon, bounding_polygon_crs)
             else:
                 spatial_extent = SpatialExtent(bounding_polygon)
+
         if len(cube.coords('time')) == 1:
-            temporal_extent = TemporalExtent(cube.coord('time').points)
-            total_temporal_extent_list.append(cube.coord('time').points)
+            time_list = num2pydate(times=cube.coord('time').points, units=cube.coord(
```
Comment (corinnebosley) [inline src/clean_air/data/extract_metadata.py:100]: As a purely aesthetic point I would prefer not to have a newline straight after an opening parenthesis, but this is not strictly necessary so still happy to approve without this change.

**Review by corinnebosley — APPROVED** [review]
Looks good to me.  

I have a couple of nitpicky comments, firstly the comment that I left in extract_metadata.py, but also generally I do like to see the purpose of the test in the description so that I know what it's trying to do.  However, as there's been a lot of work done while I've been absent there's probably going to be a lot of tests without this, so no worries.

Good job!

**Comment by lucascatherine** [comment]
Had a look through. This all looks good to me. Fine to go ahead and merge.

**Review by lucascatherine — APPROVED** [review]
Had a look through. This all looks good to me. Fine to go ahead and merge.

**Comment by tomw66** [comment]
> 

Good ideas, I've implemented both; if you want to have a quick check of the test descriptions before I merge!

File: src/clean_air/data/extract_metadata.py [hunk 2]
```diff
@@ -97,8 +97,9 @@ def extract_metadata(
                 spatial_extent = SpatialExtent(bounding_polygon)
 
         if len(cube.coords('time')) == 1:
-            time_list = num2pydate(times=cube.coord('time').points, units=cube.coord(
-                'time').units.cftime_unit, calendar=cube.coord('time').units.calendar).tolist()
+            time_list = num2pydate(times=cube.coord('time').points,
+                                   units=cube.coord('time').units.cftime_unit,
```
Comment (corinnebosley) [inline src/clean_air/data/extract_metadata.py:101]: Beautiful.

**Review by corinnebosley — APPROVED** [review]
Those descriptions are lovely.  Very concise, very clear.  

Thanks Tom, nice work!

**Pull request merged**